browserassets/js/chess.js

```diff
@@ -0,0 +1,232 @@
+// globals and chess board drawing
+var tileSize = 60;
+var files = 8, ranks = 8;
+var white = "#f0ce99", black = "#8a613f";
+var whiteSide = "white", blackSide = "black";
+var imagesPath = "{{resource('images/chessboard/)}}";
+var notationOffset = 10;
+var notationColor, notationTextStyle = "10px", weirdnessMultiplier = "5" // for whatever reason, some offsets have to be divided by this so it looks less weird. dunno why.
+
+// piece handling and selection
+var pieceList
+var fromSpace, toSpace;
+var selectionColor = "rgba(34, 139, 34, 0.7)", lastMoveColor = "rgba(255, 255, 0, 0.3)";
+var selectionSquare, lastMoveFromPos, lastMoveToPos;
+
+function byond() { // used for sending instructions to chess_board.dm
```

Comment:
honestly don't really want to merge this because it's yet another snowflake UI implementing its own byond communication protocol

there's also so much weirdness in this JS and in the byond code that would've been solved just by using tgui

Reply:
considering my lack of prior knowledge of tgui, i made this with what i was familiar and comfortable with. i had considered refactoring everything to tgui at a later date, when i would get around to it.

also, don't all the non-tgui interfaces that use browserassets define their own communication protocol like this?